cpp/src/arrow/compute/kernels/codegen_internal.h

```diff
@@ -140,6 +140,30 @@ struct GetViewType<Type, enable_if_t<is_base_binary_type<Type>::value ||
   static T LogicalValue(PhysicalType value) { return value; }
 };
 
+template <>
+struct GetViewType<Decimal32Type> {
+  using T = Decimal32;
+  using PhysicalType = std::string_view;
```

Comment:
what is the `PhysicalType` here used for compared to the `T` itself? (I was just following the existing pattern for `Decimal128/Decimal256` when I did this)

Reply:
Doing a little digging, the reason why we can't change it to `int32_t` here is because all of the decimal types are treated as `fixed_size_binary` rather than `numeric` in the rest of the codebase

Reply:
The issue comes down to `visit_data_inline.h` which uses `std::string_view` to call these `GetViewType` functions, so I don't think we could replace these `std::string_view` usages without significant refactors elsewhere.

Reply:
Yep. Some templates made the bad decision of overloading on the PhysicalType so integers can't be used as PhysicalType for other types that are not the numeric ones. This is specially problematic when trying to improve the support for half-float.

Reply:
I also think maybe int32 or `span<char, 4>` better here